composables/useEncounter.ts

```diff
@@ -0,0 +1,206 @@
+import { computed } from 'vue';
+import { useLocalStorage } from '@vueuse/core';
+import { API_ENDPOINTS, useAPI } from '~/composables/api';
+import { until } from '@vueuse/core';
+import { usePartyStore } from '~/composables/useParty';
+import { useXPCalculator } from '~/composables/useXPCalculator';
+
+interface EncounterMonster {
+  id: string;
+  name: string;
+  challenge_rating_decimal: number;
+  challenge_rating_text: string;
+  count: number;
+  // Remainder will be populated by API call
+  [key: string]: any;
+}
+
+export type { EncounterMonster };
+
+export type DifficultyLevel =
+  | 'empty'
+  | 'trivial'
+  | 'easy'
+  | 'medium'
+  | 'hard'
+  | 'deadly';
+
+export const useEncounterStore = () => {
+  const monsters = useLocalStorage<EncounterMonster[]>(
+    'encounter-monsters',
+    []
+  );
+  const monsterCache = useLocalStorage<Record<string, any>>(
+    'monster-cache',
+    {}
+  );
+  const { get } = useAPI();
+  const { partyXPBudget } = usePartyStore();
+  const xpCalculator = useXPCalculator();
+
+  const difficultyColors: Record<DifficultyLevel, string> = {
```

Comment:
I wasn't sure if this was the best place to put this, but it felt reasonable since nothing but encounters ever uses encounter difficulty.

This isn't used for 2024 rule tho, so I may refactor them out later.